sdks/java/core/src/main/java/org/apache/beam/sdk/testing/SourceTestUtils.java

```diff
@@ -673,4 +680,93 @@ public SplitAtFractionResult(int numPrimaryItems, int numResidualItems) {
         numItemsToReadBeforeSplitting, fraction, options);
     return (res.numResidualItems > 0);
   }
+
+  /**
+   * Returns an equivalent unsplittable {@code BoundedSource<T>}.
+   */
+  public static <T> BoundedSource<T> toUnsplittableSource(BoundedSource<T> boundedSource) {
```

Comment:
missing a bunch of overrides on the reader, for the functions that have default implementations.

Why is this necessary?